### PR TITLE
Relocate Share List func to modal + navigation to lists from Home + styled Home components

### LINKED ISSUE #32: Move the list sharing form to home page.
- Adding a delete button to each list.
- Moving the list sharing to home page and add it next to the delete button.
- When a user clicks on the share button a modal with the `Invite a user by email` form is displayed. 
- Clicking on a list name should direct the user to that list's page.

### PR DESCRIPTION
**Pull request by Wyna-7**
## Description

The share list functionality has been moved to a modal, which opens when the share button next to each list is clicked.
When a list name is clicked, the site navigates to the list view for that specific list.
Components visible from the home page have been styled to match Figma (as much as I could).

## Related Issue

Closes #32 

## Acceptance Criteria

- [x] Each `SingleList` component renders a button that allows the user to open a modal when clicked.
- [x] The modal allos the user to share the list with another user by filling and submitting a form.
- [x] Each `SingleList` component renders a clickable list name, which navigates to that list's view.
- [x] - [x] Each `SingleList` component renders a delete list button that does nothing (for now :))

## Type of Changes

`enhancement`

### After

image
image

## Testing Steps / QA Criteria

- Do a `git pull` and `git checkout refactor_homepage`
- Open the homepage by running `npm start`.
- The share list functionality should be in a modal, which opens clicking the share icon next to each list.
- When clicking on a list name, the site should navigate to that list's view

### CODE REVIEW

File: src/components/SingleList.jsx [hunk 1]
```diff
@@ -1,13 +1,44 @@
...
 
+	const handleShareClick = () => {
+		setSelectedItem(name);
+		handleShareModalClick();
+	};
+
 	return (
-		<li className="SingleList">
-			<button onClick={handleClick}>{name}</button>
+		<li className="flex flex-row align-middle justify-between px-2 py-1 rounded-[5px] text-[1.2em] space-x-5 w-[16rem] bg-white text-black">
```
Comment (marshjaja) [inline src/components/SingleList.jsx]:
```suggestion
		<li className="flex flex-row align-middle justify-between px-2 py-3 rounded-[5px] text-[1.2em] space-x-5 w-full bg-white text-black">
```

File: src/components/SingleList.jsx [hunk 2]
```diff
@@ -1,13 +1,44 @@
...
+				{name}
+			</button>
+			<div className="flex flex-row relative space-x-2">
+				<button
+					onClick={handleShareClick}
+					className="text-green-500 hover:text-green-600"
+				>
+					<FaShareNodes />
+				</button>
+				<button className="text-red-500  hover:text-red-600">
```
Comment (marshjaja) [inline src/components/SingleList.jsx]:
```suggestion
			<button className="text-pink-500  hover:text-pink-600">
					<RiDeleteBin5Fill size={24} />
				</button>
```

File: src/views/Home.jsx [hunk 3]
```diff
@@ -1,24 +1,46 @@
...
+				<ShareModal
+					listPath={listPath}
+					user={user}
+					name={selectedItem}
+					handleShareModalClick={handleShareModalClick}
+				/>
+			)}
 			<CreateShoppingList user={user} setListPath={setListPath} />
+			<div className="flex flex-row justify-center">
+				<ul className="flex flex-col justify-center space-y-2">
```
Comment (marshjaja) [inline src/views/Home.jsx]:
```suggestion
				<ul className="flex flex-col justify-center space-y-4 w-full max-w-xl">
```

File: src/views/Home.jsx [hunk 4]
```diff
@@ -1,24 +1,46 @@
...
-						key={index}
-						name={item.name}
-						path={item.path}
-						setListPath={setListPath}
-					/>
-				))}
-			</ul>
+		<div className="flex flex-col space-y-10 justify-center ">
+			<div className="flex flex-row justify-center">
+				<h1 className="font-[montserrat] font-bold text-[32px] underline decoration-solid decoration-4 decoration-[#2EBB4B]  ">
```
Comment (marshjaja) [inline src/views/Home.jsx]:
```suggestion
				<div className="relative inline-block">
					<h1 className=" font-[montserrat] text-3xl font-bold text-gray-300">
						<span className="relative inline-block">
							Your
							<span className="absolute bottom-[-6px] left-0 w-full h-[3px] bg-pink-500"></span>
						</span>{' '}
						Lists
					</h1>
				</div>
```

File: src/components/ShareModal.jsx [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+						<button
+							type="submit"
+							className="bg-[#FBB300] py-2 px-6 border rounded-[100px]"
+						>
+							Invite
+						</button>
+					</div>
+				</form>
+			</div>
+		</div>
```
Comment (marshjaja) [inline src/components/ShareModal.jsx]: To be more inline with the Figma design I have added these classes

```suggestion
		<div className="fixed inset-0 flex justify-center items-center z-10">
			<div
				className="fixed inset-0 bg-[rgba(49,49,49,0.8)]"
				onClick={handleShareModalClick}
			></div>
			<div>
				<form
					onSubmit={(event) => handleSubmit(event)}
					className="relative text-black bg-slate-50 z-10 rounded p-10 w-full max-w-2xl  flex flex-col justify-between items-center space-y-6"
				>
					<h2 className="font-bold text-lg">You're sharing {name}</h2>
					<div className="w-full">
						<label htmlFor="email" className="block mb-2 text-sm">
							Invite a user by email:
						</label>
						<input
							className="w-full bg-white border-[1px] border-gray-400 focus-visible:outline-none focus:ring-1 focus:ring-green-500 rounded-xl p-2"
							type="text"
							id="email"
							required
							value={recipientEmail}
							onChange={(e) => setRecipientEmail(e.target.value)}
						/>
					</div>
					<div className="flex flex-row justify-center space-x-4 w-full">
						<button
							onClick={handleShareModalClick}
							className="bg-white  hover:bg-slate-100 hover:bg-opacity-80 py-62 px-8 border rounded-xl w-1/2"
						>
							Close
						</button>
						<button
							type="submit"
							className="bg-pink-500  hover:bg-pink-500 hover:bg-opacity-80  py-3 px-8 rounded-xl w-1/2"
						>
							Invite
						</button>
					</div>
				</form>
			</div>
		</div>
		
```

File: src/components/CreateShoppingList.jsx [hunk 6]
```diff
@@ -23,16 +23,27 @@ export default function CreateShoppingList({ user, setListPath }) {
...
-				required
-			/>
-			<button type="submit">Create Shopping List</button>
+		<form onSubmit={handleSubmit} className="flex flex-row space-x-3 items-end">
+			<div className="flex flex-col">
+				<label htmlFor="shoppingList" className="text-xs font-extralight">
+					Create a new list
+				</label>
+				<input
+					className="border-[1px] rounded h-[3rem] hover:border-blue-700"
```
Comment (marshjaja) [inline src/components/CreateShoppingList.jsx]:
```suggestion
					className="border-[1px] rounded-lg h-[3rem] focus-visible:outline-none focus:ring-1 focus:ring-green-500"
```

File: src/components/CreateShoppingList.jsx [hunk 7]
```diff
@@ -23,16 +23,27 @@ export default function CreateShoppingList({ user, setListPath }) {
...
+					type="text"
+					value={listName}
+					onChange={(e) => setListName(e.target.value)}
+					required
+				/>
+			</div>
+
+			<button
+				type="submit"
+				className="bg-[#FBB300] hover:bg-[#fbb400c9] text-black font-bold h-[3rem] p-3 rounded align-bottom"
```
Comment (marshjaja) [inline src/components/CreateShoppingList.jsx]:
```suggestion
	className="bg-green-500 hover:bg-green-500 hover:bg-opacity-80 text-black font-bold h-[3rem] p-3 rounded-lg align-bottom"
```

**Review by marshjaja — CHANGES_REQUESTED** [review]
Well done! 🚀 I’ve made some suggested adjustments to align with the colour palette and design we chose on Figma.

There where a lot of file and I hope I didn't miss one😅
Please let me know if you have any questions

Screenshot 2024-10-05 at 13 49 04
Screenshot 2024-10-05 at 13 50 58

**Review by Hudamabkhoot — COMMENTED** [review]
Aside from the design notes from marcia it looks good to me great work! 👏

**Review by marshjaja — APPROVED** [review]

**Pull request merged**